### PR TITLE
feat: add popular repos query

### PR DESCRIPTION
**Pull request by bashbunni**
goal: support finding the most popular projects for an organization
used here: https://github.com/charmbracelet/.github/blob/stargazing/profile/README.md & https://github.com/charmbracelet/.github/blob/stargazing/templates/README.md.tpl#L13

todos:
- [x] update `README` with usage example for this query

### CODE REVIEW

File: repos.go [hunk 1]
```diff
@@ -69,6 +69,78 @@ var recentReleasesQuery struct {
...
+	}
+
+	for _, v := range popularReposOrgQuery.User.Organization.Repositories.Edges {
+		// ignore meta-repo
+		if string(v.Node.NameWithOwner) == fmt.Sprintf("%s/%s", username, username) {
+			continue
+		}
+
+		repos = append(repos, repoFromQL(v.Node))
+		if len(repos) == count {
```
Comment (bashbunni) [inline repos.go:123]: I think I can remove this condition since the range is already set in the query

Comment (bashbunni) [reply repos.go:123]: I do need this because of the meta repo edge case

File: repos.go [hunk 2]
```diff
@@ -69,6 +69,68 @@ var recentReleasesQuery struct {
...
+				Node qlRepository
+			}
+		} `graphql:"repositories(first: $count, privacy: PUBLIC, orderBy: {field: STARGAZERS, direction: DESC})"`
+	} `graphql:"organization(login: $orgname)"`
+}
+
+func popularRepos(count int, orgname string) []Repo {
+	fmt.Println("Finding popular repos...")
+
+	repos := make([]Repo, count)
```
Comment (bashbunni) [inline repos.go]: haha, this totally broke it last night 🙃 reverting to prev changes fixes it

File: main.go [hunk 3]
```diff
@@ -41,9 +41,10 @@ func main() {
 	/* Github */
 	funcMap["recentContributions"] = recentContributions
 	funcMap["recentPullRequests"] = recentPullRequests
-	funcMap["recentRepos"] = recentRepos
 	funcMap["recentForks"] = recentForks
 	funcMap["recentReleases"] = recentReleases
+	funcMap["recentRepos"] = recentRepos
+	funcMap["popularRepos"] = popularRepos
```
Comment (ccoVeille) [inline main.go:47]: You said you were hesitating about the name

Maybe popularOrgRepos, or popularReposOrg (see https://github.com/charmbracelet/markscribe/pull/5#discussion_r1673076207)

You said it's limited to an organization

File: repos.go [hunk 4]
```diff
@@ -69,6 +69,68 @@ var recentReleasesQuery struct {
...
+        cursor
+        node {
+          nameWithOwner
+        }
+      }
+    }
+  }
+}*/
+
+var popularReposOrgQuery struct {
```
Comment (ccoVeille) [inline repos.go]:
```suggestion
var popularOrgReposQuery struct {
```

It sounds better to me, but I might be wrong

**Review by ccoVeille — COMMENTED** [review]
Thanks for bringing this feature.

**Comment by bashbunni** [comment]
I thiiink this would be better as a method for the Org struct. More legible + organized, so that'll be the next step here

it would look something like:
`{{.Org.PopularRepos "charmbracelet" 10}}`, then the names can be simplified. A lot of this was originally written for a single user, would be nice to separate the org-specific functions this way

File: README.md [hunk 5]
```diff
@@ -120,6 +120,24 @@ Stars: {{.Stargazers}}
 This function requires GitHub authentication with the following API scopes:
 `repo:status`, `public_repo`, `read:user`.
 
+### Repositories with the most stars
+
+```
+{{range popularRepos "charmbracelet" 10}}
```
Comment (ccoVeille) [inline README.md:126]: Shouldn't it be?


```suggestion
{{range orgPopularRepos "charmbracelet" 10}}
```

Comment (bashbunni) [reply README.md:126]: Thank you!!

**Comment by bashbunni** [comment]
fwiw my original proposal to make these functions easier to reuse between user and organization will take some refactoring, but for right now we just need it to work. Would be fun to refactor this down the line though

**Comment by bashbunni** [comment]
You can see the results of this version in this pull request: https://github.com/charmbracelet/.github/pull/6/files#diff-0e83982cf6f4dabedcbf7b12029a56c6f4a728064ba99543fff8f227c0654b45R12-R22

File: main.go [hunk 6]
```diff
@@ -41,9 +41,10 @@ func main() {
 	/* Github */
 	funcMap["recentContributions"] = recentContributions
 	funcMap["recentPullRequests"] = recentPullRequests
-	funcMap["recentRepos"] = recentRepos
 	funcMap["recentForks"] = recentForks
 	funcMap["recentReleases"] = recentReleases
+	funcMap["recentRepos"] = recentRepos
+	funcMap["orgPopularRepos"] = orgPopularRepos
```
Comment (aymanbagabas) [inline main.go]: This is not specific for organizations though, you should be able to use it for users as well. I would rename it to `popularRepos`

Comment (aymanbagabas) [reply main.go]: Or `userPopularRepos`

Comment (bashbunni) [reply main.go]: Ah hold on, I finally figured out how to make this work for users and orgs 🎊  I'll update the query + solution across the board here. Thanks for the nudge :)

**Review by aymanbagabas — APPROVED** [review]
LGTM, great work @bashbunni !

**Pull request merged**